src/Processors/Chunk.h

```diff
@@ -202,6 +204,9 @@ class Chunk
     void append(Chunk && chunk);
 
     /// proton : starts
+    /// Clone chunk with copyed chunk context, used for shuffling to prevent access to the same chunk context
```

Comment:
I think this method is generic and don't necessary get used only by light shuffling.